pkg/client/create_builder.go

```diff
@@ -323,6 +361,44 @@ func (c *Client) addConfig(ctx context.Context, kind string, config pubbldr.Modu
 	return nil
 }
 
+func (c *Client) processBuilderCreateTargets(ctx context.Context, opts CreateBuilderOptions) ([]dist.Target, error) {
+	var targets []dist.Target
+
+	// For backward compatibility, we must create a target with de default platform used in the Fetch implementation for daemon and remote
+	// When daemon
+	//		OS:           daemonInfo.Os,
+	//		Architecture: daemonInfo.Arch
+	// When remote || layout
+	// 		OS:           "linux",
+	//		Architecture: runtime.GOARCH
```

Comment:
Is this still accurate? What about the build-time base image os/arch?

Reply:
At this point, we haven't fetched the build-time image. we changed the logic the builder creation logic to be:
```
For each t given `target`
  createBuilder(t)
```  
When end-users don't provided any target (flag or configuration) we will create a default one for them to execute the iteration, in those cases, that is how we are initializing the `os` and `arch`

Reply:
I think when user doesn't provide any targets we should just assume single os/arch and grab the values from the build image

Reply:
Is that not how it works today?

Reply:
I did the following:
- When user doesn't provide any targets, the method `processBuilderCreateTargets` returns an array with a empty target `dist.Target{}` I need something in the array to actually run the iteration at least one
- During the **loop** I am checking if the target is empty, in such as case, I sent `nil` to the creation methods, which in theory are running the same logic we have today.

Reply:
Why not have a check for `if len(targets) == 0 { <it should follow the old code path> }`?

Reply:
I did that, let's check how it looks now.

Reply:
LGTM!